static/js/popovers.js

```diff
@@ -684,6 +684,7 @@ export function toggle_actions_popover(element, id) {
             should_display_reminder_option: feature_flags.reminders_in_message_action_menu,
             should_display_edit_and_view_source,
             should_display_quote_and_reply,
+            not_spectator: !page_params.is_spectator,
```

Comment:
I suppose the other approach here would be to have all of these `should_display` properties add an `&& !page_params.is_spectator` in the JavaScript.  I kinda prefer that approach for readability, just because every one of these options has a `should_display_` parameter, and those parameters should semantically control what happens.

And I think ultimately that'll be more flexible, if we later decide we want to offer "View source" or something in this menu.